src/sdp/media/ersip_sdp_attr_rtpmap.erl

```diff
@@ -0,0 +1,209 @@
+
+-export([new/3, new/4]).
+
+-export([payload_type/1]).
+-export([encoding_name/1]).
+-export([clock_rate/1]).
+-export([encoding_params/1]). 
+-export([set_payload_type/2]).
+-export([set_encoding_name/2]).
+-export([set_clock_rate/2]).
+-export([set_encoding_params/2]).
+
+-export([parse/1]).
+-export([assemble/1]).
+-export([assemble_bin/1]).
+
+-define(IS_DIGIT(X),    (X >= $0 andalso X =< $9)).
+-define(IS_CAP_CHAR(X), (X >= $A andalso X =< $Z)).
+-define(IS_LOW_CHAR(X), (X >= $a andalso X =< $z)).
+-define(IS_LETTER(X),   (?IS_CAP_CHAR(X) orelse ?IS_LOW_CHAR(X) orelse X == $- )).
+
+%%%===================================================================
+%%% Types
+%%%===================================================================
+
+-record(rtpmap, {
+    type           :: payload_type(),
+    encoding_name  :: encoding_name(),
+    clock_rate     :: clock_rate(),
+    params         :: encoding_params() 
+}).
+
+-type payload_type()     :: binary().
+-type encoding_name()    :: binary().
+-type clock_rate()       :: pos_integer().
+-type encoding_params()  :: pos_integer() | undefined.
+-type rtpmap()           :: #rtpmap{}.
+-type parse_result()     :: parse_result(rtpmap()).
+-type parse_result(T)    :: ersip_parser_aux:parse_result(T).
+
+-export_type([
+    payload_type/0,
+    clock_rate/0,
+    encoding_name/0    
+]).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+-spec new(payload_type(), encoding_name(), clock_rate()) -> rtpmap().
+new(Type, Name, ClockRate) ->
+    new(Type, Name, ClockRate, undefined).
+
+-spec new(payload_type(), encoding_name(), clock_rate(), encoding_params()) -> rtpmap().
+new(Type, Name, ClockRate, Params) ->
+    #rtpmap{
+        type = Type,
+        encoding_name = Name,
+        clock_rate = ClockRate,
+        params = Params
+    }.
+
+-spec payload_type(rtpmap()) -> payload_type().
+payload_type(#rtpmap{type = Type}) ->
+    Type.
+
+-spec encoding_name(rtpmap()) -> encoding_name().
+encoding_name(#rtpmap{encoding_name = Name}) ->
+    Name.
+
+-spec clock_rate(rtpmap()) -> clock_rate().
+clock_rate(#rtpmap{clock_rate = ClockRate}) ->
+    ClockRate.
+
+-spec encoding_params(rtpmap()) -> encoding_params().
+encoding_params(#rtpmap{params = Params}) ->
+    Params.
+
+-spec set_payload_type(payload_type(), rtpmap()) -> rtpmap().
+set_payload_type(Type, Rtpmap) ->
+    Rtpmap#rtpmap{type = Type}.
+
+-spec set_encoding_name(encoding_name(), rtpmap()) -> rtpmap().
+set_encoding_name(Name, Rtpmap) ->
+    Rtpmap#rtpmap{encoding_name = Name}.
+
+-spec set_clock_rate(clock_rate(), rtpmap()) -> rtpmap().
+set_clock_rate(ClockRate, Rtpmap) ->
+    Rtpmap#rtpmap{clock_rate = ClockRate}.
+
+-spec set_encoding_params(encoding_params(), rtpmap()) -> rtpmap().
+set_encoding_params(Params, Rtpmap) ->
+    Rtpmap#rtpmap{params = Params}.
+
+-spec parse(binary()) -> parse_result().
+parse(Bin) ->
```

Comment:
If this function is supposed to be called to result value from attr list then there is no sense to return Rest there. Non-empty rest means parse error.

Reply:
Last notice regarding to MR is to check that Rest is empty. Current parser would accept something like this:
`a=rtpmap:106 OPUS/48000/2 Some Garbage`

Reply:
Add handler for non empty Rest.